pkg/funky/server.go

```diff
@@ -96,10 +96,10 @@ func (s *DefaultServer) Invoke(input *Message) (interface{}, error) {
 	p, err := json.Marshal(input)
 
 	var timeout time.Duration
-	if (input.Context.Deadline == time.Time{}) {
+	if (*input.Context.Deadline == time.Time{}) {
```

Comment:
`omitempty` setting may result in `nil` value for `Deadline`, this could be simplified to:
```go
timeout := time.Duration(0) 
if input.Context.Deadline != nil {
    timeout = time.Until(*input.Context.Deadline)
}
```